### PR TITLE
Able to load a uds file to pointCloud

### PR DESCRIPTION
**Pull request by kressnick25**
Implemented loading `uds` file to `pointCloud` with file selector

Co-authored by @jmkraut 
## Other notable changes
- Handled errors better using exceptions by extending a parent UdSDKClass which all udSDK classes extend and therefore share the same error handling.

- Pointers and allocations now handled internally by each UdSDK class so as to avoid doing memory management outside lib. Just have to call cleanup() to free allocated memory. just have to be careful not to allow object to be de-referenced before calling cleanup() as this will leak memory. 

- Fixed the launch config so don't have to launch twice
### Home page
Screenshot_20200827-154227
### Select file screen
Screenshot_20200827-154236
### Error message for bad file format
Can be changed to more descriptive, this is just to show it works and doesn't crash the application on failure
Screenshot_20200827-154250

### CODE REVIEW

File: vaultSDK/lib/udAttributes.dart [hunk 1]
```diff
@@ -0,0 +1,730 @@
...
+  }
+
+  /// Gets the offset for a standard attribute so that further querying of that attribute can be performed
+  ///
+  /// @param pAttributeSet The attribute set to get the offset for
+  /// @param attribute The enum value of the attribute
+  /// @param pOffset This pointer will be written to with the value of the offset if it is found
+  /// @return A udError value based on the result of writing the offset to pOffset
+  void getOffsetOfNamedAttribute(String name, int offset) {
+    final Pointer<Uint32> pOffset = allocate();
```
Comment (samsinsane) [inline vaultSDK/lib/udAttributes.dart:52]: You don't call `free` for this.

Comment (kressnick25) [reply vaultSDK/lib/udAttributes.dart:52]: Are we doing this completely wrong then? I just looked at what the function does and I'm thinking we should either:
- return an `int` from the wrapper function `getOffsetNamedAttribute`
- leave the return type as `void` and have `Pointer<Uint32> pOffset` as a property of the class

Comment (samsinsane) [reply vaultSDK/lib/udAttributes.dart:52]: My apologies, I didn't look very closely at the function itself, just that it differed to the one above. Yes, this should return an `int` (the offset), so it should look something like:
```dart
int getOffsetOfNamedAttribute(String name) {
  final Pointer<Uint32> pOffset = allocate();
  final pName = Utf8.toUtf8(name);
  handleUdError(_udAttributeSet_GetOffsetOfNamedAttribute(
      _attributeSet, pName, pOffset));
  final offset = pOffset.value;
  free(pName);
  free(pOffset);
  return offset;
}
```

Comment (samsinsane) [reply vaultSDK/lib/udAttributes.dart:52]: Also, to clarify, the function above this one should be adjusted similarly.

Comment (kressnick25) [reply vaultSDK/lib/udAttributes.dart:52]: Updated now 262d68d

**Pull request merged**